uplink/config.go

```diff
@@ -41,10 +43,47 @@ type RSConfig struct {
 // EncryptionConfig is a configuration struct that keeps details about
 // encrypting segments
 type EncryptionConfig struct {
-	Key       string      `help:"root key for encrypting the data"`
-	BlockSize memory.Size `help:"size (in bytes) of encrypted blocks" default:"1KiB"`
-	DataType  int         `help:"Type of encryption to use for content and metadata (1=AES-GCM, 2=SecretBox)" default:"1"`
-	PathType  int         `help:"Type of encryption to use for paths (0=Unencrypted, 1=AES-GCM, 2=SecretBox)" default:"1"`
+	// TODO: WIP#if/v3-1541 read TODO in Key method why the following field cannot
+	// be added
+	// RootKey     *storj.Key  // The root key for encrypting the data and read it from KeyFilepath, see Key method.
+	KeyFilepath string      `help:"the path to the file which contains the root key for encrypting the data"`
+	BlockSize   memory.Size `help:"size (in bytes) of encrypted blocks" default:"1KiB"`
+	DataType    int         `help:"Type of encryption to use for content and metadata (1=AES-GCM, 2=SecretBox)" default:"1"`
+	PathType    int         `help:"Type of encryption to use for paths (0=Unencrypted, 1=AES-GCM, 2=SecretBox)" default:"1"`
+}
+
+// LoadKey returns the root key for encrypting the data which is stored in the path
+// indicated by KeyFilepath field.
+//
+// It returns an error if:
+//
+// * The file doesn't exist.
+// * There is an I/O error.
+//
+// TODO: WIP#if/v3-1541 refactor cache the key after the firs read. The problem
+// is with pkg/cfgstruct which is inflexible on accepting unexported keys nor
+// several different types as a type of storj.Key, [32]byte, etc.
+func (cfg *EncryptionConfig) LoadKey() (key storj.Key, err error) {
```

Comment:
I'm wondering if the returned key should be a pointer because I read in a few places that we want to keep the minimum copies of them in memory as possible.

I mostly applied such logic, however I missed this one.